### PR TITLE
Falcon40b - Enable SDPA as single op

### PR DESCRIPTION
**Pull request by djordje-tt**
### Ticket
- https://github.com/tenstorrent/tt-metal/issues/9637

### Problem description
- Scaled dot product attention was implemented as sequence of ops and now there is support as single op. Falcon40b can benefit from it and utilising it improves both device and e2e time.
### What's changed
- All ops that represented sdpa are not replaced with single op call
- Code is simpler and perf is optimized

### Checklist
- [ ] demo tests: https://github.com/tenstorrent/tt-metal/actions/runs/9658846609
- [ ] unit tests: https://github.com/tenstorrent/tt-metal/actions/runs/9658857175
- [ ] frequent tests: https://github.com/tenstorrent/tt-metal/actions/runs/9658865126
- [ ] model perf tests: https://github.com/tenstorrent/tt-metal/actions/runs/9658878866

### CODE REVIEW

**Comment by ayerofieiev-tt** [comment]
> and utilising it improves both device and e2e time.

Do we have good tools available to show perf impact?

**Comment by djordje-tt** [comment]
> > and utilising it improves both device and e2e time.
> 
> Do we have good tools available to show perf impact?

@ayerofieiev-tt, for e2e perf we calculate python time for e2e execution and for device time we are using tracy to capture 1 layer perf and compared sdpa op with all that are replaced with.

File: tests/scripts/t3000/run_t3000_model_perf_tests.sh [hunk 1]
```diff
@@ -66,7 +66,7 @@ run_t3000_llm_tests() {
   run_t3000_mixtral_tests
 
   # Run llama2-70b tests
-  run_t3000_llama2_70b_tests
+  # run_t3000_llama2_70b_tests
```
Comment (uaydonat) [inline tests/scripts/t3000/run_t3000_model_perf_tests.sh:69]: why is the test removed?

Comment (djordje-tt) [reply tests/scripts/t3000/run_t3000_model_perf_tests.sh:69]: It's only temporary commented for model perf test to hit falcon40b. That is the easiest way to hit models that are not executing due to other models failing at the moment. Before merging all the changes inside tests are discarded and tests remain the same. Also, llama2 team is informed about failure on CI.

**Review by johanna-rock-tt — APPROVED** [review]
Looks good to me if all CI pipelines pass.

**Review by TT-BrianLiu — APPROVED** [review]

**Comment by ttmchiou** [comment]
linked frequent pipeline shows llama tests? I don't see any falcon40b tests run
also it looks like falcon40b model perf tests timed out at 60 minutes?

**Comment by ttmchiou** [comment]
there are new changes in t3k pipelines so would suggest rerunning for any issues

**Comment by djordje-tt** [comment]
@ttmchiou, yes. I found an issue with sdpa which I am currently debugging. Once I fix it, I will rerun all pipelines definitely.

File: tests/scripts/t3000/run_t3000_model_perf_tests.sh (hunk 1 above)
Comment (ttmchiou) [reply tests/scripts/t3000/run_t3000_model_perf_tests.sh:85]: please revert back and recomment these tests.
Llama tests are now split so this type of change shouldn't be required to test

File: tests/scripts/t3000/run_t3000_unit_tests.sh [hunk 2]
```diff
@@ -109,7 +109,7 @@ run_t3000_tests() {
   run_t3000_ttmetal_tests
 
   # Run ttnn tests
-  run_t3000_ttnn_tests
+  # run_t3000_ttnn_tests
```
Comment (ttmchiou) [inline tests/scripts/t3000/run_t3000_unit_tests.sh:112]: Please revert

**Comment by johanna-rock-tt** [comment]
@djordje-tt While you're actively working on an issue you can convert the PR to a draft, then people know that the current version of your branch is not supposed to be merged to main and still WIP.

**Pull request closed without merging**